internal/auth/header_builder.go

```diff
@@ -0,0 +1,115 @@
+// It defines options for signing requests and selecting the appropriate key for signing.
+type HeaderConfig struct {
+	SignRequest bool               // SignRequest indicates whether to sign the request.
+	PrivateKey  *ec.PrivateKey     // PrivateKey is used for signing when no ExtendedKey is provided.
+	ExtendedKey *bip32.ExtendedKey // ExtendedKey is used for signing when provided.
+}
+
+// IsExtendedKey checks if an extended key is available for signing.
+func (h HeaderConfig) IsExtendedKey() bool { return h.ExtendedKey != nil }
+
+// IsPrivateKey checks if a private key is available for signing.
+func (h HeaderConfig) IsPrivateKey() bool { return h.PrivateKey != nil }
+
+// IsSignRequest indicates whether the request should be signed based on
+// the configuration.
+func (h HeaderConfig) IsSignRequest() bool { return h.SignRequest }
+
+// HeaderBuilder constructs HTTP headers for signing requests.
+// It supports building headers using either an ExtendedKey (bip32.ExtendedKey)
+// or a PrivateKey (ec.PrivateKey). The header can include a signature
+// based on the specified Body and the SignRequest flag.
+type HeaderBuilder struct {
+	cfg *HeaderConfig
+}
+
+// Build generates the HTTP headers based on the keys provided.
+// It returns an error if neither an ExtendedKey nor a PrivateKey is set.
+// Body is the content to be signed or included in the header.
+func (b *HeaderBuilder) Build(body string) (http.Header, error) {
+	if b.cfg.IsExtendedKey() {
+		return b.buildWithExtendedKey(body)
+	}
+	if b.cfg.IsPrivateKey() {
+		return b.buildWithPrivateKey(body)
+	}
+	return nil, ErrMissingKeys
+}
+
+// BuildWithoutBody constructs an HTTP header without any request body content.
+// This method calls the Build method with an empty string as the body, creating
+// a header suitable for GET, DELETE, or other HTTP requests that do not require
+// a request payload.
+func (b *HeaderBuilder) BuildWithoutBody() (http.Header, error) {
+	return b.Build("")
+}
+
+// buildWithExtendedKey creates headers using the ExtendedKey.
+// It returns an error if the ExtendedKey is not set or if
+// there is an issue generating the public key or signature.
+func (b *HeaderBuilder) buildWithExtendedKey(body string) (http.Header, error) {
+	header := make(http.Header)
+	xPub, err := bip32.GetExtendedPublicKey(b.cfg.ExtendedKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get extended public key: %w", err)
+	}
+	header.Set(models.AuthHeader, xPub)
+
+	if b.cfg.IsSignRequest() {
+		if err := setSignature(&header, b.cfg.ExtendedKey, body); err != nil {
+			return nil, fmt.Errorf("failed to set signature: %w", err)
+		}
+	}
+	return header, nil
+}
+
+// buildWithPrivateKey creates headers using the PrivateKey.
+// It returns an error if the PrivateKey is not set or if
+// there is an issue generating the signature for the access key.
+func (b *HeaderBuilder) buildWithPrivateKey(body string) (http.Header, error) {
```

Comment:
The `PrivateKey` can lead the programmers to think, that it is used also for `xPrivs` 
- but actually it is only for `access keys`

Reply:
Any proposition?

Reply:
I suppose, after rework based on the above ☝️ comment - this one solves automatically.